burn-core/src/nn/norm/group.rs

```diff
@@ -0,0 +1,179 @@
+use crate as burn;
+
+use crate::config::Config;
+use crate::module::Module;
+use crate::module::Param;
+use crate::tensor::backend::Backend;
+use crate::tensor::Tensor;
+
+/// Configuration to create a [GroupNorm](GroupNorm) layer.
+#[derive(Config)]
+pub struct GroupNormConfig {
+    /// The number of groups to separate the channels into
+    num_groups: usize,
+    /// The number of channels expected in the input
+    num_channels: usize,
+    /// A value required for numerical stability. Default: 1e-5
+    #[config(default = 1e-5)]
+    epsilon: f64,
+    /// A boolean value that when set to `true`, this module has learnable
+    /// per-channel affine parameters initialized to ones (for weights)
+    /// and zeros (for biases). Default: `true`
+    #[config(default = true)]
+    affine: bool,
+}
+
+/// Applies Group Normalization over a mini-batch of inputs.
+///
+/// `Y = groupnorm(X) * γ + β`
+#[derive(Module, Debug)]
+pub struct GroupNorm<B: Backend> {
+    num_groups: usize,
+    num_channels: usize,
+    gamma: Param<Tensor<B, 1>>,
+    beta: Param<Tensor<B, 1>>,
```

Comment:
I would wrap those in `Option` so when group norm is configured with `afine=false` no weights are added!